### DISCUSSION TITLE
New audio player

### DISCUSSION THREAD

**Discussion by tphakala** [tphakala/birdnet-go#285] [Announcements]
Container images contain a new audio player which is now integrated into spectrogram image, player controls are hidden until mouse is moved on top of spectrogram image. You can also skip within audio clip by clicking on spectrogram.

https://github.com/user-attachments/assets/e5d0b04a-537d-40b6-9ae1-3834e64ef46c

There are still few todo's for audio player
- add export function to specific format, if audio is saved as flac you could export audio as mp3 for example
- allow selecting a part of audio only for export
- add optional noise suppression using ffmpeg filters for playback and export

**Comment by vibragiel** [comment]
Hi @tphakala! A very happy user here. This new player looks gorgeous. However, it isn't working for me. It does something, and I can see it in the inspector, but something seems broken. It's happening both in Firefox and Chrome (Fedora Linux in both cases).


https://github.com/user-attachments/assets/1ebace4a-f678-4f35-84dc-9b1d9403c41d

**Reply by tphakala** [reply]
That's odd, I see that audio clip is there as otherwise spectrogram wouldn't been generated. I also noticed you have wav as audio export format, that version you are using also now supports aac (m4a), opus, flac and mp3, how about if you switch format to aac for example, does it behave same?

**Reply by tphakala** [reply]
And btw, does console on that browser developer mode show any errors?

**Reply by vibragiel** [reply]
Just changed it to AAC and after waiting for a new detection, the behavior is the same.

> And btw, does console on that browser developer mode show any errors?

Yes it does, though it occurs on entering the dashboard, and it seems to work fine (audio player included, which seems to be the legacy one in the dashboard?).

![image](https://github.com/user-attachments/assets/5a0b8402-cc91-445d-a266-e54a80bae75b)

When I drill down to a specific species it's when the app tries to show the new player and fails, but no new errors occur in the console.

**Reply by tphakala** [reply]
I created issue https://github.com/tphakala/birdnet-go/issues/287 about this, lets continue discussion over there
